experimental/fairseq_moe_docker_slurm/run_fairseq_moe_aml.py

```diff
@@ -0,0 +1,136 @@
+    '--decoder-attention-heads', 32,
+    '--moe-expert-count', NUM_EXPERTS, 
+    '--moe-freq', 2,
+    '--moe-gating-use-fp32',
+    '--moe-second-expert-policy', 'all',
+    '--moe-normalize-expert-grad', 'sqrt_world_size',
+    '--moe-eval-capacity-token-fraction', -1.0,
+    '--max-sentences-valid', 1, 
+    '--num-workers-valid', 0,
+    '--criterion', 'moe_cross_entropy', 
+    '--moe-gate-loss-wt', 0.01, 
+    '--moe-gate-loss-combine-method', 'sum',
+    '--optimizer', 'adam', 
+    '--fp16-adam-stats', 
+    #'--adam-betas', (0.9,0.98), 
+    '--clip-norm', 0.0,
+    '--lr', 0.0005, 
+    '--warmup-updates', 750,
+    '--dropout', 0.1, 
+    '--attention-dropout', 0.1,
+    '--batch-size', BATCH_SIZE, 
+    '--update-freq', GRAD_ACC,
+    '--max-update', MAX_UPDATE, 
+    '--disable-validation',
+    '--log-format', 'json', 
+    '--log-interval', 10,
+]
+
+#-------------------------------------------------------------------------------
+# Create ScriptRunConfig
+#-------------------------------------------------------------------------------
+distr_config = PyTorchConfiguration(node_count=2)
+launch_cmd = "cd /workspace/fairseq && \
+        python -m torch.distributed.launch --nproc_per_node 8 --nnodes 2 \
+        --node_rank $NODE_RANK --master_addr $MASTER_ADDR --master_port $MASTER_PORT \
+        train.py \
+        --save-dir ./ \
+        --ddp-backend fully_sharded --memory-efficient-fp16 --checkpoint-activations \
+        --task dummy_lm --tokens-per-sample 1024 \
+        --arch transformer_lm_gpt --share-decoder-input-output-embed \
+        --decoder-layers 32 --decoder-embed-dim 4096 --decoder-ffn-embed-dim 16384 \
+        --decoder-attention-heads 32 \
+        --moe-expert-count 8 --moe-freq 2 \
+        --moe-gating-use-fp32 --moe-second-expert-policy all \
+        --moe-normalize-expert-grad sqrt_world_size \
+        --moe-eval-capacity-token-fraction -1.0 \
+        --max-sentences-valid 1 --num-workers-valid 0 \
+        --criterion moe_cross_entropy --moe-gate-loss-wt 0.01 --moe-gate-loss-combine-method sum \
+        --optimizer adam --fp16-adam-stats --adam-betas '(0.9,0.98)' --clip-norm 0.0 \
+        --lr 0.0005 --warmup-updates 750 \
+        --dropout 0.1 --attention-dropout 0.1 \
+        --batch-size 6 --update-freq 2 \
+        --max-update 25 --disable-validation \
+        --log-format json --log-interval 10".split()
+
+fairseq_src = ScriptRunConfig(
+    source_directory='/path/to/your/code/files',
+    #script='train.py',
+    #arguments=run_args,
+    command=launch_cmd,
+    compute_target=compute_target,
+    environment=fairseq_env,
+    distributed_job_config=distr_config
+    )
+
+#-------------------------------------------------------------------------------
+# Submit experiment
+#-------------------------------------------------------------------------------
+# Name your experiment by setting YOUR_EXPERIMENT_NAME
+run = Experiment(ws, 'YOUR_EXPERIMENT_NAME').submit(fairseq_src)
```

Comment:
I'd suggest adding CLI args (e.g. from `argparse` or `click`) that can be passed into a main function instead of having to edit this script for different experiments.

Reply:
Good point. Thanks!